sql/moz-fx-data-shared-prod/accounts_backend_external/nonprod_accounts_v1/schema.yaml

```diff
@@ -26,6 +29,9 @@ fields:
 - name: profileChangedAt
   type: NUMERIC
   mode: NULLABLE
+- name: keysChangedAt
+  type: NUMERIC
```

Comment:
I'd strongly recommend converting the timestamp and boolean columns into the appropriate native BigQuery type for usability.

Looks like the timestamp column values are milliseconds-since-Unix-epoch, so that'd be `SAFE.TIMESTAMP_MILLIS(SAFE_CAST(... AS INT))`, and the boolean columns would be `SAFE_CAST(... AS BOOL)`.

Reply:
I'm wondering if it's better to keep them as is in the table to:
1. Preserve original format to make it less surprising for FxA developers looking at this table
2. Make it easier to switch to some other CDC tool (e.g. Dataform) in case we decide it's preferable instead of federated queries.

For other consumers we could do the conversions you proposed in the view.

What do you think?

Reply:
That approach would be OK, just less convenient IMO.

Personally I wouldn't worry too much about FxA developers being surprised.  If anything, I'd think they'd be _pleasantly_ surprised they didn't have to figure out that `TIMESTAMP_MILLIS()` is the BigQuery function to convert milliseconds into a usable timestamp value, and then deal with the confusion of not being able to pass the `NUMERIC` millisecond values directly to that function because it expects an `INT64`.

And I wouldn't be worried about switching to some other CDC tool, because in that case the data would probably be landing in some other project and we'd have rote copying ETLs here where we could apply similar data type conversions.

Reply:
Makes sense, I'll convert these in the table.